include/deal.II/fe/fe_coupling_values.h

```diff
@@ -0,0 +1,1724 @@
+ * const auto right = fe_coupling.right(scalar);
+ *
+ * // Now we can use the augmented extractors to access the values of the two
+ * // FEValuesBase objects
+ * const auto & left_vi = fe_coupling[left].value(i, q);
+ * const auto & right_vi = fe_coupling[right].value(i, q);
+ * @endcode
+ */
+template <int dim1, int dim2 = dim1, int spacedim = dim1>
+class FECouplingValues
+{
+public:
+  /**
+   * Construct the FECouplingValues in an invalid state. Before you can use this
+   * object, you must call the reinit() function.
+   */
+  FECouplingValues();
+
+  /**
+   * Construct the FECouplingValues with two arbitrary FEValuesBase objects.
+   * This class assumes that the FEValuesBase objects that are given at
+   * construction time are initialized and ready to use (i.e., that you have
+   * called the reinit() function on them before calling this constructor).
+   *
+   * @param fe_values_1 The left FEValuesBase object.
+   * @param fe_values_2 The right FEValuesBase object.
+   * @param dof_coupling_type The type of dof coupling to use.
+   * @param quadrature_coupling_type The type of quadrature to use for the coupling.
+   */
+  FECouplingValues(
+    const FEValuesBase<dim1, spacedim> &fe_values_1,
+    const FEValuesBase<dim2, spacedim> &fe_values_2,
+    const DoFCouplingType &dof_coupling_type = DoFCouplingType::independent,
+    const QuadratureCouplingType &quadrature_coupling_type =
+      QuadratureCouplingType::tensor_product);
+
+  /**
+   * Reinitialize the FECouplingValues with two arbitrary FEValuesBase objects.
+   * The FEValuesBase objects must be initialized and ready to use, i.e., you
+   * must have called the reinit() function on them before calling this method.
+   *
+   * @param fe_values_1 The left FEValuesBase object.
+   * @param fe_values_2 The right FEValuesBase object.
+   * @param dof_coupling_type The type of dof coupling to use.
+   * @param quadrature_coupling_type The type of quadrature to use for the coupling.
+   */
+  void
+  reinit(
+    const FEValuesBase<dim1, spacedim> &fe_values_1,
+    const FEValuesBase<dim2, spacedim> &fe_values_2,
+    const DoFCouplingType &dof_coupling_type = DoFCouplingType::independent,
+    const QuadratureCouplingType &quadrature_coupling_type =
+      QuadratureCouplingType::tensor_product);
+
+  /**
+   * Helper struct to associate an extractor to the left FEValuesBase object.
+   */
+  template <typename Extractor>
+  struct LeftCoupling
+  {
+    LeftCoupling(const Extractor &extractor)
+      : extractor(extractor)
+    {}
+
+    /**
+     * The actual extractor object.
+     */
+    const Extractor extractor;
+  };
+
+  /**
+   * Helper struct to associate an extractor to the right FEValuesBase object.
+   */
+  template <typename Extractor>
+  struct RightCoupling
+  {
+    RightCoupling(const Extractor &extractor)
+      : extractor(extractor)
+    {}
+
+    /**
+     * The actual extractor object.
+     */
+    const Extractor extractor;
+  };
```

Comment:
These two classes look identical!? What is the reason?

Reply:
They are used to identify which of the two `FEValuesBase` you are addressing. Asking for `fev[left]` and `fev[right]` will have a different effect.